cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -29,6 +36,19 @@ import java.util.*
  * All methods called within the [FlowLogic] sub-class need to be annotated with the @Suspendable annotation.
  */
 object PaybackFlow {
+    fun findMoney(serviceHub: ServiceHub, party: Party): StateAndRef<CashState> {
+        // looking for money that should be printed on ExampleFlow
+        // this is not the best way, but I use this way to exercise, I will do better in module 4
+        val criteria = QueryCriteria.VaultQueryCriteria(
+                contractStateTypes = setOf(CashState::class.java),
+                status = Vault.StateStatus.UNCONSUMED,
+                exactParticipants = listOf(party)
+        )
+
+        val results = serviceHub.vaultService.queryBy(IOUState::class.java, criteria)
+        return serviceHub.toStateAndRef<CashState>(results.states.single().ref)
```

Comment:
Why not return the list because you don't know what is the sum of what you got from the vault.